### PR TITLE
Deploy Frontend and Backend Resources

### PR DESCRIPTION
**Pull request by sienkin**
Added initial TF config to deploy Frontend and Backend resources for WCK in Heritage Dev, Staging and Live

Partially Resolves: CM-1161

### CODE REVIEW

File: groups/wck-infrastructure/alb_external.tf [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+# External ALB CloudWatch Merics
+#--------------------------------------------
+module "external_alb_metrics" {
+  source = "git@github.com:companieshouse/terraform-modules//aws/alb-metrics?ref=tags/1.0.26"
+
+  load_balancer_id = module.wck_external_alb.this_lb_id
+  target_group_ids = module.wck_external_alb.target_group_arns
+
+  depends_on = [module.wck_external_alb]
+}
```
Comment (MartyFox) [inline groups/wck-infrastructure/alb_external.tf]: Could you update this to the latest version of the ALB Alarms?
Example available here:
`https://github.com/companieshouse/ceu-terraform/blob/main/groups/ceu-frontend/alb-internal.tf`

File: groups/wck-infrastructure/alb_internal.tf [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    map(
+      "ServiceTeam", "${upper(var.application)}-FE-Support"
+    )
+  )
+}
+
+#--------------------------------------------
+# Internal ALB CloudWatch Merics
+#--------------------------------------------
+module "internal_alb_metrics" {
```
Comment (MartyFox) [inline groups/wck-infrastructure/alb_internal.tf]: Same as above

File: groups/wck-infrastructure/asg_backend.tf [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+  user_data_base64               = data.template_cloudinit_config.bep_userdata_config.rendered
+
+  tags_as_map = merge(
+    local.default_tags,
+    map(
+      "ServiceTeam", "${upper(var.application)}-BEP-Support"
+    )
+  )
+}
+
```
Comment (MartyFox) [inline groups/wck-infrastructure/asg_backend.tf:109]: Could you also add the ASG alarms to this?
Example here:
`https://github.com/companieshouse/ceu-terraform/blob/main/groups/ceu-frontend/asg_frontend.tf`

File: groups/wck-infrastructure/asg_frontend.tf [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+    map(
+      "ServiceTeam", "${upper(var.application)}-FE-Support"
+    )
+  )
+
+  depends_on = [
+    module.wck_external_alb,
+    module.wck_internal_alb
+  ]
+}
```
Comment (MartyFox) [inline groups/wck-infrastructure/asg_frontend.tf:124]: Same as above for ASG Alarms

**Review by MartyFox — COMMENTED** [review]
I've left a few inline comments, both are CloudWatch alarm related and will require SNS as well, example for this can be found here:
`https://github.com/companieshouse/ceu-terraform/blob/main/groups/ceu-frontend/sns.tf`

**Comment by sienkin** [comment]
> I've left a few inline comments, both are CloudWatch alarm related and will require SNS as well, example for this can be found here: `https://github.com/companieshouse/ceu-terraform/blob/main/groups/ceu-frontend/sns.tf`

SNS and CloudWatch alarm configuiration added in [4c83e30](https://github.com/companieshouse/wck-terraform/pull/1/commits/4c83e306127c1bbb5979beab7029feee829e44a5)

File: groups/wck-infrastructure/alb_external.tf [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+    map(
+      "ServiceTeam", "${upper(var.application)}-FE-Support"
+    )
+  )
+}
+
+#--------------------------------------------
+# External ALB CloudWatch Merics
+#--------------------------------------------
+module "external_alb_metrics" {
```
Comment (MartyFox) [inline groups/wck-infrastructure/alb_external.tf]: This also needs replaced to match the updated internal alarms.

Comment (sienkin) [reply groups/wck-infrastructure/alb_external.tf]: @MartyFox Updated in [0909666](https://github.com/companieshouse/wck-terraform/pull/1/commits/0909666d71fb830fa059bf468bfc60aada2baf35)

**Review by MartyFox — APPROVED** [review]

**Pull request merged**